internal/terminal/table_test.go

```diff
@@ -0,0 +1,272 @@
+
+import (
+	"fmt"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/fatih/color"
+	"testing"
+)
+
+func TestEmptyTableConstructor(t *testing.T) {
+	for _, tc := range []struct {
+		description string
+		header      []string
+		data        []map[string]interface{}
+	}{
+		{
+			description: "Should return empty table if there is nil for the header",
+			header:      nil,
+			data:
+			[]map[string]interface{}{
+				{"test": "should not show up"},
+			},
+		},
+		{
+			description: "Should return empty table if there is no header information",
+			header:      []string{},
+			data: []map[string]interface{}{
+				{"test": "should not show up"},
+			},
+		},
+	} {
+		t.Run(tc.description, func(*testing.T) {
+			table := newTable(tc.header, tc.data)
+			assert.Equal(t, []string{}, table.headers)
+			assert.Equal(t, []map[string]string{}, table.data)
+			assert.Equal(t, map[string]int{}, table.columnWidths)
+		})
+	}
+}
+
+func TestTableConstructor(t *testing.T) {
+	t.Run("Should return a table with only headers if there is one and no data", func(t *testing.T) {
+		headers := []string{"should", "show", "up", "only"}
+		table := newTable(headers, nil)
+		assert.Equal(t, table.headers, headers)
+		assert.Equal(t, []map[string]string{}, table.data)
+		assert.Equal(t, table.columnWidths, map[string]int{
+			"should": 6,
+			"show":   4,
+			"up":     2,
+			"only":   4,
+		})
+	})
+	t.Run("Should return a table if there is both a header and data", func(t *testing.T) {
+		table := newTable(
+			[]string{"test"},
+			[]map[string]interface{}{
+				{
+					"test":   "should show up",
+					"int":    1,
+					"float":  1.000,
+					"float2": 1.234567890123,
+					"array":  []string{"1", "2", "3"},
+					"nil":    nil,
+				},
+			},
+		)
+		assert.Equal(t, table.headers, []string{"test"})
+		assert.Equal(t, table.data, []map[string]string{
+			{
+				"test":   "should show up",
+				"int":    "1",
+				"float":  "1",
+				"float2": "1.234567890123",
+				"array":  "[1 2 3]",
+				"nil":    "",
+			},
+		})
+	})
+}
+
+func TestTable_Message(t *testing.T) {
+	t.Run("Should return an empty string and an error if the table has no headers", func(t *testing.T) {
+		data := []map[string]interface{}{
+			{"test": 123, "this": "456", "data": []string{"7", "8 9", "10!"}},
+		}
+		table := newTable(nil, data)
+		message, error := table.Message()
+		assert.Equal(t, message, "")
+		assert.Equal(t, error.Error(), "cannot create a table without headers")
+	})
+
+	t.Run("Should print only a header for a table with no data", func(t *testing.T) {
+		table := newTable([]string{"header", "only", "no", "data"}, nil)
+		message, error := table.Message()
+		assert.Nil(t, error)
+		assert.Equal(t, message, "\n"+fmt.Sprintf("%s  %s  %s  %s",
+			[]interface{}{
+				color.New(color.Bold).SprintFunc()("header"),
+				color.New(color.Bold).SprintFunc()("only"),
+				color.New(color.Bold).SprintFunc()("no"),
+				color.New(color.Bold).SprintFunc()("data"),
```

Comment:
For all the occurrences of `"data"` and `"header"` wouldn't it be better to use the `const` values in `table.go` in the tests?

Reply:
Most of these occurrences of "data" and "header" are really values within the actual header that just so happened to be named "header" and "data"

I'll change it for the `Payload` test though